### PR TITLE
Add a Fedora cOS variant

### LINKED ISSUE #61: cOS variants
It is possible already to specify a different base image in https://github.com/mudler/cOS/blob/master/packages/base/definition.yaml#L6 but it's not sufficient to build `cOS` on top of a `foo` distribution.

We have already a mechanism for templating globally, that can be exploited to provides the values in the specific templates ( e.g. https://github.com/mudler/cOS/tree/master/values ) .

The Issue is about creating a new `cOS` variant (ubuntu, debian, etc) , and attaching to the `CI`and build it aside the standard `cOS`. 

The values files in `values` corresponds to a specific variant which is being targeted while running `make build`. For now, we are sticking with a generic "amd64" which is empty only because the mechanism can be used also to provide multi-arch  builds, but we can reuse the same mechanism to provide different `cOS` flavors. 

See also: https://luet-lab.github.io/docs/docs/concepts/packages/templates/#interpolating-globally

### PR DESCRIPTION
**Pull request by davidcassany**
This commit adds a variant for cOS based on Fedora. Small refactor in
Makefile has been done to allow different flavors. Each flavor maps to a
set to an specific yaml file including the required additional values.
The default flavor is set to opensuse.

In addition the kernel package has been dropped and included as part of
the base distro package, this is done to facilitate templating across
different distros.

Finally there is an unrelated update for the mount point of the ephemeral
overlay volume that gets mounted in /run/overlay now instead of
/data.overlay.

Fixes #61

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -43,23 +43,24 @@ endif
 endif
 
 clean:
-	 rm -rf $(ROOT_DIR)/build/ $(ROOT_DIR)/.qemu
+	 rm -rf $(DESTIBNATION) $(ROOT_DIR)/.qemu
```
Comment (mudler) [inline Makefile:46]:
```suggestion
	 rm -rf $(DESTINATION) $(ROOT_DIR)/.qemu
```

File: Makefile [hunk 2]
```diff
@@ -13,9 +12,10 @@ export TREE?=$(ROOT_DIR)/packages
 BUILD_ARGS?=--pull --no-spinner --only-target-package
 
 VALIDATE_OPTIONS?=-s
-ARCH?=amd64
-REPO_CACHE?=raccos/$(ARCH)
-FINAL_REPO?=raccos/releases-$(ARCH)
+FLAVOR?=opensuse
```
Comment (mudler) [inline Makefile:15]: :+1:

File: Makefile [hunk 3]
```diff
@@ -80,30 +81,30 @@ publish-repo:
     --type docker
 
 serve-repo:
-	LUET_NOLOCK=true $(LUET) serve-repo --port 8000 --dir $(ROOT_DIR)/build
+	LUET_NOLOCK=true $(LUET) serve-repo --port 8000 --dir $(DESTIONATION)
```
Comment (mudler) [inline Makefile:84]:
```suggestion
	LUET_NOLOCK=true $(LUET) serve-repo --port 8000 --dir $(DESTINATION)
```

**Review by mudler — APPROVED** [review]
Looks good! it seems although that the ci fails for the missing `values` files

I'll try to hook later the CI to build ISO/images for the fedora spin

**Comment by mudler** [comment]
Just a later thought: we have also to configure the upgrade channel somehow, at the moment is configured in the cloud-config package, but would make sense to let it configure in the `values`

File: Makefile (hunk 1 above)
Comment (davidcassany) [reply Makefile:46]: ouch :man_facepalming:  fixed now

File: Makefile (hunk 3 above)
Comment (davidcassany) [reply Makefile:84]: Oh man, I solved the conflicts on Makefile too fast :sweat:

**Comment by davidcassany** [comment]
> Looks good! it seems although that the ci fails for the missing `values` files
> 
> I'll try to hook later the CI to build ISO/images for the fedora spin

Damn I forgot to commit the values files :sweat:

File: Makefile [hunk 4]
```diff
@@ -13,9 +12,10 @@ export TREE?=$(ROOT_DIR)/packages
 BUILD_ARGS?=--pull --no-spinner --only-target-package
 
 VALIDATE_OPTIONS?=-s
-ARCH?=amd64
-REPO_CACHE?=raccos/$(ARCH)
-FINAL_REPO?=raccos/releases-$(ARCH)
+FLAVOR?=opensuse
+DESTINATION?=$(ROOT_DIR)/build
```
Comment (davidcassany) [inline Makefile:16]: JFYI I'd would have made the this line like
```
DESTINATION?=$(ROOT_DIR)/build-$(FLAVOR)
```
So I could easily build the iso from opensuse or fedora without having to rebuild the repository. However the geniso extension is dependent on the  `$(ROOT_DIR)/build` path. No big deal, but something to keep in mind for #56

Comment (mudler) [reply Makefile:16]: Is geniso using it explictly or it's caused by https://github.com/mudler/cOS/blob/master/iso/cOS.yaml#L20 ?

Comment (davidcassany) [reply Makefile:16]: Oh, I guess you are right, I did not look into that. I saw it was failing in one of the geniso scripts but I did not check where the path was defined.

**Pull request merged**